ARMmbed/mbedtls
# Remove HAVE_RAM_AVAILABLE_128K in favour of dynamic heap checking in test

## Description
Ref. email discussions with @ronald-cron-arm and @gilles-peskine-arm. `HAVE_RAM_AVAILABLE_128K` was put in as a hasty fix, and doesn't scale that well (i.e. what about 64K? 32K?). It's definition was also pretty unclear: is it 128K total? Or 128K heap size?

Since the test system now got extended with `ASSERT_ALLOC_WEAK`, we can leverage that specifically for those tests which try to exercise the absolute largest key size possible. If allocating a buffer the size of the key fails, then the test is skipped, instead of requiring the user to know about a well-hidden compile-time define that needed to be commented out manually in the test files.

## Status
**READY**

## Requires Backporting
NO  

## Migrations
NO

## Todos

## Steps to test or reproduce
No impact on testing. All tests should still succeed on regular targets (with large enough memory sizes). This change enables on-target testing for constrained devices with the same set of tests as otherwise run by default.
